packages/mattermost-redux/src/selectors/entities/channel_categories.ts

```diff
@@ -468,14 +469,31 @@ export function makeSortChannels() {
     };
 }
 
-export function makeGetChannelsForCategory() {
+export function makeGetChannelIdsForCategory() {
     const getChannels = makeGetChannelsForIds();
     const filterAndSortChannelsForCategory = makeFilterAndSortChannelsForCategory();
+    let prevChannels: Channel[] = [];
+    let prevFilteredChannelIds: string[] = [];
 
     return (state: GlobalState, category: ChannelCategory) => {
         const channels = getChannels(state, category.channel_ids);
 
-        return filterAndSortChannelsForCategory(state, channels, category);
+        if (channels.length === 0 && prevChannels.length === 0) {
+            return prevFilteredChannelIds;
+        }
+
+        // getChannels will almost never return a memoized result.
+        // Assumption here is that the added comparison computation is more beneficial
+        // than the cost of returning a non memoized result.
+        if (channels.length === prevChannels.length &&
```

Comment:
It looks like `fast-deep-equal` already does a length check: 

https://github.com/epoberezkin/fast-deep-equal/blob/a8e7172b6c411ec320d6045fd4afbd2abc1b4bde/src/index.jst#L56-L59

And it also optimizes for checking the array in reverse (probably assuming that items are changed there first). Does it make sense to simplify this to just the `deepEqual` call?

Reply:
You know what, I should have checked that. I'll remove the manual length check 👍